### PR TITLE
refactor(form-field): remove 6.0.0 deletion targets

### PR DESCRIPTION
**Pull request by crisbeto**
Removes the targets marked for 6.0.0 deletion from the `material/form-field`.

BREAKING CHANGES:
* `dividerColor` which was deprecated in 5.0.0 has been removed. Use `color` instead.
* `floatPlaceholder` which was deprecated in 5.0.0 has been removed. Use `floatLabel` instead.
* `MatFormFieldControl.shouldPlaceholderFloat` which was deprecated in 5.0.0 has been removed. Use `shouldLabelFloat` instead.
* `FloatPlaceholderType`  which was deprecated in 5.0.0 has been removed. Use `FloatLabelType` instead.
* `PlaceholderOptions` which was deprecated in 5.0.0 has been removed. Use `LabelOptions` instead.
* `MAT_PLACEHOLDER_GLOBAL_OPTIONS`  which was deprecated in 5.0.0 has been removed. Use `MAT_LABEL_GLOBAL_OPTIONS` instead.
* The `mat-input-placeholder-wrapper` and `mat-form-field-placeholder-wrapper` CSS classes which were deprecated in 5.0.0 have bee removed. The same element can be targeted using the `mat-form-field-label-wrapper` class.
* The `mat-input-placeholder` and `mat-form-field-placeholder` CSS classes which were deprecated in 5.0.0 have bee removed. The same element can be targeted using the `mat-form-field-label` class.

### CODE REVIEW

File: src/lib/form-field/form-field.html [hunk 1]
```diff
@@ -8,21 +8,10 @@
     <div class="mat-input-infix mat-form-field-infix" #inputContainer>
       <ng-content></ng-content>
 
-      <!--
-        TODO: remove `mat-input-placeholder-wrapper` and `mat-form-field-placeholder-wrapper`
-        next time we do breaking changes.
-        @deletion-target 6.0.0
-      -->
-      <span class="mat-form-field-label-wrapper mat-input-placeholder-wrapper mat-form-field-placeholder-wrapper">
+      <span class="mat-form-field-label-wrapper">
```
Comment (mmalerba) [inline src/lib/form-field/form-field.html]: Can you add a `@deletion-target` while you're here to remove all of the `mat-input-*` classes? We can do the actual remove in a different PR because I know for sure it's going to be a nightmare to merge

Comment (crisbeto) [reply src/lib/form-field/form-field.html]: Done. I've set it to 7.0.0, but I can bump it down to 6.0.0 if you want.

**Comment by jelbourn** [comment]
Can you change the breaking notes to match that earlier PR where applicable (i.e. "Xxx, which was deprecated in 5.0.0, has been removed. Use yyy instead")

**Comment by crisbeto** [comment]
Done.

**Comment by jelbourn** [comment]
Caretaker note: we need to remove google usages of the old API. This might require a cleanup before syncing this PR.

**Comment by mmalerba** [comment]
closing since all shard PRs have been merged
